e2e/tests/attendance-tests.spec.ts

```diff
@@ -1,360 +1,278 @@
 import { expect, test } from "#e2e/fixtures.ts";
 
-test.describe("Attendance Module", () => {
-  const FIXED_DATE = "2025-01-23";
+test.only("Record attendance for one activity", async ({ page }) => {
+  await page.getByRole("navigation").getByText("Attendance").click();
+  await page.getByRole("button", { name: "Record" }).click();
 
-  test.beforeEach(async ({ page }) => {
-    await page.clock.install({ time: FIXED_DATE });
-  });
+  const dateField = page.getByLabel("Date");
+  await expect(dateField).toHaveValue("1/23/2025");
 
-  test("Record attendance for one activity", async ({ page }) => {
-    await page.getByRole("navigation").getByText("Attendance").click();
-    await page.getByRole("button", { name: "Record" }).click();
+  await dateField.fill("12/15/2024");
 
-    /*
-      Verify the date field displays the current date
-    */
-    const fixedDate = new Date(FIXED_DATE);
-    const formattedDateRegex = new RegExp(
-      `${fixedDate.getMonth() + 1}/${fixedDate.getDate()}/${fixedDate.getFullYear()}|` +
-        `${(fixedDate.getMonth() + 1).toString().padStart(2, "0")}/${fixedDate.getDate().toString().padStart(2, "0")}/${fixedDate.getFullYear()}`,
-    );
-    const dateField = page.getByLabel("Date");
-    await expect(dateField).toHaveValue(formattedDateRegex);
+  // FIXME: A simple .click() does not trigger the action and we don’t know why.
+  await page.getByText("School Class 2B").dispatchEvent("click");
 
-    /*
-      Allow backdated editing for the date field
-    */
-    const backdatedDate = "12/15/2024";
-    await dateField.fill(backdatedDate);
-    await expect(dateField).toHaveValue(backdatedDate);
+  await expect(
+    page.getByRole("heading", { name: "School Class 2B" }),
+  ).toBeVisible();
 
-    /*
-      Verify list of classes is displayed
-    */
-    await page.waitForSelector("mat-card-header");
-    const classList = page.locator("mat-card-header");
-    const count = await classList.count();
-    expect(count).toBeGreaterThan(0);
+  await expect(page.getByText("1 / 6")).toBeVisible();
+  await page.getByRole("button", { name: "Present" }).click();
 
-    /*
-      Mark attendance for participants dynamically
-    */
-    const firstClass = page.locator("mat-card-header").first();
-    await firstClass.evaluate((el: HTMLElement) => el?.click()); // Force the DOM click
-
-    let totalParticipants = 0;
-
-    const pageIndicator = page.locator("text=/\\d+\\s\\/\\s\\d+/"); // "1 / 5" pattern
-    // Extract total participants from "x / y" format
-    if (await pageIndicator.isVisible()) {
-      const text = await pageIndicator.textContent();
-      const match = text?.match(/(\d+)\s\/\s(\d+)/); // Extract current/total numbers
-      if (match) {
-        totalParticipants = parseInt(match[2]); // Total participants
-      }
-    }
```

Comment:
In general we should avoid making the test code dependent of the fixtures in the database. Since we setup the tests in a specific state we know the participant count and can assert it directly:

```typescript
await expect(page.getByText("1 / 6")).toBeVisible();
```